pkg/apis/policy/types.go

```diff
@@ -444,3 +449,18 @@ type PodSecurityPolicyList struct {
 
 	Items []PodSecurityPolicy
 }
+
+// RuntimeClassStrategyOptions define the strategy that will dictate the allowable RuntimeClasses
+// for a pod.
+type RuntimeClassStrategyOptions struct {
+	// AllowedRuntimeClasses is a whitelist of RuntimeClass names that may be specified on a pod.
+	// An empty string value allows the RuntimeClassName to be left unset (or set to "").
+	// An empty list allows all RuntimeClasses.
+	// +optional
+	AllowedRuntimeClasses []string
+	// DefaultRuntimeClass is the default RuntimeClassName to set on the pod.
+	// The default MUST be allowed by the above AllowedRuntimeClasses.
+	// A value of nil does not mutate the Pod.
+	// +optional
+	DefaultRuntimeClass *string
```

Comment:
**Question:** Should we omit this? I'm generally against defaulting in the PodSecurityPolicy, but since the PSP already does defaulting this seems more consistent.

We have also discussed adding a RuntimeClass label selector to the pod, in which case a RuntimeClass admission controller would pick a RuntimeClass value. That would be problematic with setting a default here, depending on how the exact ordering works (do built-in controllers run before or after webhooks?)

Reply:
I think we should stick with the default here (for consistency). If we add the label selection logic, then PSP should be updated to skip defaulting when a label selector is defined.